### PR TITLE
Switch to GH actions

### LINKED ISSUE #340: Switch to GH actions
Travis is going away for open-source projects, so CI will have to be migrated to GitHub actions.
If you agree, I'm ready to help.

### PR DESCRIPTION
**Pull request by levonet**
close #340

Removed the travis config and added the same test action to Github Action.
Added macOS platform for run tests.
I didn't manage to run tests on the Windows platform. Too many errors running tests. I can't fix this because I'm not familiar with windows.

To add this Github workflow you should add the Codecov token to Github secrets. Follow this instruction:
1. Get your Codecov token on the Codecov [setting page](https://app.codecov.io/gh/pyparsing/pyparsing/settings).
2. Set your Codecov token inside Github _Settings > Secrets_ as `CODECOV_TOKEN`.

Example of run this workflow https://github.com/levonet/pyparsing/actions/runs/1539173383

### CODE REVIEW

**Comment by ptmcg** [comment]
Ok, we will give it a shot.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tox codecov
```
Comment (ptmcg) [inline .github/workflows/ci.yml:49]: Add dependencies for railroad testing, either:

    pyparsing[railroad]

or

    jinja
    railroad-diagrams

**Comment by ptmcg** [comment]
Great, thanks!

**Comment by levonet** [comment]
Thanks, it works https://github.com/pyparsing/pyparsing/actions/runs/1545384563

**Pull request merged**